### DISCUSSION TITLE
Run VuePress dev mode

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13705] [Q&A] [answered]
I cloned the repositories titanium-mobile and titanium-docs

```
/repos$ ls -la
total 16
drwxr-xr-x  4 mudde mudde 4096 Jan  3 14:49 .
drwxr-xr-x  3 mudde mudde 4096 Jan  3 14:45 ..
drwxr-xr-x  8 mudde mudde 4096 Jan  6 01:13 titanium-docs
drwxr-xr-x 16 mudde mudde 4096 Jan  3 14:47 titanium_mobile

```

Then I install dependencies under `./titanium-docs` and tried to run VuePress in dev mode:

```
/repos/titanium-docs$ npm run docs:dev

> titanium-docs@1.0.0 docs:dev
> vuepress dev docs .temp


 WAIT  Extracting site metadata...
Error: [vuepress] Failed to load custom theme "titanium". File vuepress-theme-titanium/Layout.vue does not exist.
    at resolveOptions (/home/user/Documents/myTitanium/titanium-docs/node_modules/vuepress/lib/prepare/resolveOptions.js:69:15)
    at prepare (/home/user/Documents/myTitanium/titanium-docs/node_modules/vuepress/lib/prepare/index.js:11:25)
    at dev (/home/user/Documents/myTitanium/titanium-docs/node_modules/vuepress/lib/dev.js:23:25)
    at /home/user/Documents/myTitanium/titanium-docs/node_modules/vuepress/bin/vuepress.js:105:12
    at Command.<anonymous> (/home/user/Documents/myTitanium/titanium-docs/node_modules/vuepress/bin/vuepress.js:32:21)
    at Command.listener (/home/user/Documents/myTitanium/titanium-docs/node_modules/commander/index.js:315:8)
    at Command.emit (node:events:513:28)
    at Command.parseArgs (/home/user/Documents/myTitanium/titanium-docs/node_modules/commander/index.js:651:12)
    at Command.parse (/home/user/Documents/myTitanium/titanium-docs/node_modules/commander/index.js:474:21)
    at Object.<anonymous> (/home/user/Documents/myTitanium/titanium-docs/node_modules/vuepress/bin/vuepress.js:97:9)

``` 

If I go to `./titanium-docs/node_modules/vuepress-theme-titanium` there is no file `Layout.vue`. But in the subdirectory `vuepress-theme-titanium/layouts/Layout.vue`.

When I tried to build the docs , then I also ran into error:

```
/repos$ $ npm run docs:metadata
npm ERR! Missing script: "docs:metadata"
npm ERR! 
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2023-01-06T00_20_37_469Z-debug-0.log

```

**Comment by m1ga** [comment, marked as answer]
I've followed the steps in https://github.com/tidev/titanium-docs#installation on an older Macbook Air and it worked right away. 
node 16.18.0 and the current repo. No complaints about missing files after running `npm run docs:dev`, just the localhost webserver with the files.
Did `npm i` inside the titanium-docs folder end with errors? Try deleting the folder and clone the repo again.

**Reply by beppo-ivel** [reply]
> I've followed the steps in https://github.com/tidev/titanium-docs#installation 

I followed exactly same instructions:

```
git clone https://github.com/tidev/titanium-docs.git
git clone https://github.com/tidev/titanium-mobile.git
cd titanium-docs
npm i
npm run docs:dev
```


> node 16.18.0 and the current repo.

Using node 16.19 could that be the problem?
```
$ node -v
v16.19.0

```
> Did `npm i` inside the titanium-docs folder end with errors?

not errors but 129 vulnerabilities.

> Try deleting the folder and clone the repo again.

I deleted the node_modules directory inside repo and execute `npm i` again. Is that not the same?

**Reply by m1ga** [reply]
![Screenshot_20230106_122641](https://user-images.githubusercontent.com/4334997/211003631-f075c17b-6788-4514-833b-22077bd864da.png)

**Reply by beppo-ivel** [reply]
Ok thx recloned repo worked. Now I have same score of vulnerabilities like you.

**Reply by m1ga** [reply]
but did the other parts work too?

**Reply by beppo-ivel** [reply]
`npm run docs:dev` is working as expected. To push my changes to main I can follow the commands under [Deploy](https://github.com/tidev/titanium-docs#deploy)?

**Reply by m1ga** [reply]
no, you can just work in a branch and then make a PR to the repo. I think that deploy part is for publishing it to the domain. That's handled by a maintainer. You follow the [contribution](https://github.com/tidev/titanium-docs#contributions) part. Make the PR and then you can check the CLA stuff

**Reply by cb1kenobi** [reply]
Create a pull request and we can review it. Upon merge, it is automatically deployed!
